opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -47,6 +47,7 @@ type GeneralConfig struct {
 	PluginsList    []string `json:"pluginsList,omitempty"`
 	// Additional volumes to mount to all pods in the cluster
 	AdditionalVolumes []AdditionalVolume `json:"additionalVolumes,omitempty"`
+	Monitoring        bool               `json:"monitoring,omitempty"`
```

Comment:
I would prefer for this to be an object where one field is a boolean `enabled` field. Gives us more flexibility to make the monitoring configurable by the user.

Reply:
I planed to do it anyway to add configurations, will add it now